src/Umbraco.Core/Runtime/SqlMainDomLock.cs

```diff
@@ -0,0 +1,407 @@
+﻿using System;
+using System.Data;
+using System.Data.SqlClient;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Persistence.Dtos;
+using Umbraco.Core.Persistence.Mappers;
+using Umbraco.Core.Persistence.SqlSyntax;
+
+namespace Umbraco.Core.Runtime
+{
+    internal class SqlMainDomLock : IMainDomLock
```

Comment:
yes but remember this service/object is created pre-bootup since it's required by bootup and at that state there is no UmbracoDatabase, repositories, services, etc... so We can't really get our normal KeyValueRepository into this class without service locators or something.